src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +37,126 @@
+
+    /**
+     * Checks whether the given input is in the correct format or non-empty.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @return Original input.
+     * @throws ParseException if the given {@code jobTitle} is invalid.
+     */
+    public String checkNotEmptyJobTitle(String jobTitle) throws ParseException {
+        requireNonNull(jobTitle);
+        String trimmedJobTitle = jobTitle.trim();
+        if (!JobTitle.isValidJobTitle(trimmedJobTitle)) {
+            throw new ParseException(JobTitle.MESSAGE_CONSTRAINTS);
+        }
+        return trimmedJobTitle;
     }
 
+    /**
+     * Checks whether the given input is in the correct format or non-empty.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @return Original input.
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public String checkNotEmptyApplicationStatusTag(String tag) throws ParseException {
+        requireNonNull(tag);
+        String trimmedTag = tag.trim().toUpperCase();
+        if (!Tag.isValidTagName(trimmedTag)) {
+            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        }
+        if (!ApplicationStatusTagType.contains(trimmedTag)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddCommand.MESSAGE_APPLICATION_STATUS_TAG));
+        }
+        return trimmedTag;
+    }
+
+    /**
+     * Checks whether the given input is in the correct format or non-empty.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @return Original input.
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public String checkNotEmptyPriorityTag(String tag) throws ParseException {
+        requireNonNull(tag);
+        String trimmedTag = tag.trim().toUpperCase();
+        if (!Tag.isValidTagName(trimmedTag)) {
+            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        }
+        if (!PriorityTagType.contains(trimmedTag)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddCommand.MESSAGE_PRIORITY_TAG));
+        }
+        return trimmedTag;
+    }
+
+    /**
+     * Checks whether the given input is in the correct format or non-empty.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @return Original input.
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public Collection<String> checkNotEmptyTags(Collection<String> tags) throws ParseException {
+        requireNonNull(tags);
+        if (tags.isEmpty() || (tags.size() == 1 && tags.iterator().next().isEmpty())) {
+            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        }
+        for (String tagName : tags) {
+            if (!Tag.isValidTagName(tagName)) {
+                throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+            }
+        }
+        return tags;
+    }
```

Comment:
Perhaps for this method you want to create another MESSAGE_MUST_NOT_BE_EMPTY String in Tag.java class?

Because the MESSAGE_CONSTRAINTS would not be the same for "add" and "edit" command.

Reply:
The t/ accepting empty inputs seems to be a bug from AB3 and is not a feature for our application, and as discussed with the team we are proceeding with the check to prevent empty t/ inputs for add and find commands